### PR TITLE
cleanup and rewrite some URIs

### PR DESCRIPTION
**Pull request by TheRealBro**
- cleaning up some URIs

### CODE REVIEW

**Review by michaelstingl — CHANGES_REQUESTED** [review]
don't change the `redirect_uris`. This are the real ones, hardcoded in our mobile clients

**Comment by TheRealBro** [comment]
> don't change the `redirect_uris`. This are the real ones, hardcoded in our mobile clients

done, please re-review

**Review by michaelstingl — APPROVED** [review]

**Comment by xoxys** [comment]
@phil-davis @micbar @jnweiger any objections or can we go forward and merge it?

File: compose/kopano/konnect/konnectd-identifier-registration.yaml [hunk 1]
```diff
@@ -18,15 +18,15 @@ clients:
...
 
   - id: mxd5OQDk6es5LzOzRvidJNfXLUZS2oN3oUFeXPP8LpPrhx3UroJFduGEYIBOxkY1
     secret: KFeFWWEZO9TkisIQzR3fo7hfiMXlOpaqP8CFuTbSHzV1TUuGECglPxpiVKJfOXIx
     name: ownCloud iOS app
     application_type: native
     redirect_uris:
-      - oc://ios.owncloud.com
-      - oc.ios://ios.owncloud.com
+      - oc://ios.example.com
+      - oc.ios://ios.example.com
```
Comment (jnweiger) [inline compose/kopano/konnect/konnectd-identifier-registration.yaml]: This too. 
These redirect_uris, are names that must not change.

File: compose/kopano/konnect/docker-compose.yml [hunk 2]
```diff
@@ -39,8 +39,8 @@ services:
       - signing_private_key=/kopano/ssl/konnectd-tokens-signing-key.pem
       - validation_keys_path=/kopano/ssl/konnectkeys
       - KONNECT_BACKEND=ldap
-      - LDAP_BASEDN=dc=owncloudqa,dc=com
-      - LDAP_BINDDN=cn=admin,dc=owncloudqa,dc=com
+      - LDAP_BASEDN=dc=example,dc=com
+      - LDAP_BINDDN=cn=admin,dc=example,dc=com
```
Comment (jnweiger) [inline compose/kopano/konnect/docker-compose.yml:43]: Will that still work? 
My LDAP knowledge is too little to judge that.

File: examples/hetzner-deploy/README.md [hunk 3]
```diff
@@ -9,16 +9,16 @@ Not tested on Mac. (Pull requests welcome)
...
 
 ```
 sudo apt install git
 git clone https://github.com/owncloud-docker/compose-playground
 cd compose-playground/examples/hetzner-deploy
 
-export HCLOUD_SSHKEY_NAMES=jw@owncloud.com
-export HCLOUD_TOKEN=mZdZX......................................................L8bml
+export HCLOUD_SSHKEY_NAMES=user@example.com
+export HCLOUD_TOKEN=asdf......................................................12345
```
Comment (jnweiger) [inline examples/hetzner-deploy/README.md]: That change is okay. It is not code. It is an example in a README.

File: examples/hetzner-deploy/lib/hcloud_py/make_machine.py [hunk 4]
```diff
@@ -129,7 +129,7 @@ def ssh_fingerprint(str):
 IPADDR = server.data_model.public_net.ipv4.ip
 print("Machine created: %s" % IPADDR, file=sys.stderr)
 
-server.change_dns_ptr(IPADDR, NAME+'.hcloud.owncloud.com')       # needs an FQDN
+server.change_dns_ptr(IPADDR, NAME+'.hcloud.example.com')       # needs an FQDN
```
Comment (jnweiger) [inline examples/hetzner-deploy/lib/hcloud_py/make_machine.py]: Maybe valid? DNS expert needed to review.

File: examples/hetzner-deploy/make_oc10_apps.sh [hunk 5]
```diff
@@ -206,7 +206,7 @@ for param in \$PARAM; do
 	occ config:system:set trusted_domains 2 --value="\$wopi_fqdn"
 	echo >> ~/POSTINIT.msg "RICHDOCUMENTS: The following manual steps are needed to use richdocuments"
 	echo >> ~/POSTINIT.msg "RICHDOCUMENTS:  - Register at dash.cloudflare.com:  $IPADDR \$wopi_fqdn"
-	echo >> ~/POSTINIT.msg "RICHDOCUMENTS:  - To get a certificate, run:        certbot -m qa@owncloud.com --no-eff-email --agree-tos -d \$wopi_fqdn"
+	echo >> ~/POSTINIT.msg "RICHDOCUMENTS:  - To get a certificate, run:        certbot -m qa@example.com --no-eff-email --agree-tos -d \$wopi_fqdn"
```
Comment (jnweiger) [inline examples/hetzner-deploy/make_oc10_apps.sh]: I've confirmed I can generate a cert with that emai.
The change implies, that letsencrypt cannot contact us. The man page says
`-m EMAIL         Email address for important account notifications`

Would it violate any rules?

File: examples/hetzner-deploy/oidc_login/start_client.sh [hunk 6]
```diff
@@ -28,7 +28,7 @@ shift; shift
 shift; shift
 
 test -z "$client_name" && client_name=owncloud
-test -z "$server_url"  && server_url=https://demo.owncloud.com
+test -z "$server_url"  && server_url=https://demo.example.com
```
Comment (jnweiger) [inline examples/hetzner-deploy/oidc_login/start_client.sh]: a non-working default, is acceptable here for now. The script is not used currently.

**Review by jnweiger — CHANGES_REQUESTED** [review]
Added several comments inline.

I am certain, the indetifier-regstration.yml will break. For most others, I don't have the expertise.
To be safe, I'd need to test all the changes. Please assign a priority if that needs to be done.
I currently am blocked with P1 and P2 items.

**Comment by TheRealBro** [comment]
Short bump here, this is really important. I'm willing to work on the issues that @jnweiger pointed out... but we still lead people to setup their domain on Let's Encrypt with our mail address... bad behaviour

**Comment by micbar** [comment]
@jnweiger you requested changes, please unblock the change if possible

**Comment by micbar** [comment]
@TheRealBro This branch needs a rebase

File: examples/hetzner-deploy/make_oc10_apps.sh (hunk 5 above)
Comment (TheRealBro) [reply examples/hetzner-deploy/make_oc10_apps.sh]: well you should 1st not use productive Let's Encrypt servers for testing see:
-  https://letsencrypt.org/docs/staging-environment/
Please use their staging environment

Second - you should have a private copy of this repo containing your private information - like mail addresses strings etc.

File: examples/hetzner-deploy/oidc_login/start_client.sh (hunk 6 above)
Comment (TheRealBro) [reply examples/hetzner-deploy/oidc_login/start_client.sh]: Same as above... private data in private repo please.... can be based on this repo.

**Comment by TheRealBro** [comment]
@micbar hetzner-deploy folder seems to have moved away to a QA repo.
Rebase done.
I guess all conflicts are solved... @jnweiger can you give a thumbs up?

**Comment by TheRealBro** [comment]
sorry... i missed one conflict... working on it

**Comment by TheRealBro** [comment]
I guess we are finally good to go :)

**Review by jnweiger — APPROVED** [review]
LGTM

example.com replaces owncoudqa.com -- I cannot directly see if that causes issues somewhere.

**Pull request merged**